### PR TITLE
pinokio: init at 1.3.4

### LINKED ISSUE #320847: Package request: pinokio
**Project description**
The elevator pitch:
```
Pinokio is a browser that lets you install, run, and programmatically control ANY application, automatically.
```

But it's mostly used to install and run AI applications pretty easily, the creator made this video: https://youtu.be/mB_bZC8pMDA
Which is the same as on the homepage.

**Metadata**

* homepage URL: https://pinokio.computer/
* source URL: https://github.com/pinokiocomputer/pinokio
* license: MIT
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ByteSudoer**
## Description of changes
Pinokio is a browser that lets you install, run, and programmatically control ANY application, automatically.
Metadata
- homepage URL: https://pinokio.computer/
- source URL: https://github.com/pinokiocomputer/pinokio
- license: MIT
- platforms: linux,darwin

Closes https://github.com/NixOS/nixpkgs/issues/320847

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bbigras** [comment]
How do you control a gui app with that?

File: pkgs/by-name/pi/pinokio/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+let
+  pname = "pinokio";
+  version = "1.3.4";
+  src = fetchurl {
+    x86_64-darwin = {
+      url = "https://github.com/pinokiocomputer/${pname}/releases/download/${version}/Pinokio-${version}.dmg";
+      hash = "sha256-Il5zaVWu4icSsKmMjU9u1/Mih34fd+xNpF1nkFAFFGo=";
+    };
+    x86_64-linux = {
+      url = "https://github.com/pinokiocomputer/${pname}/releases/download/${version}/Pinokio-${version}.AppImage";
```
Comment (SuperSandro2000) [inline pkgs/by-name/pi/pinokio/package.nix:16]:
```suggestion
      url = "https://github.com/pinokiocomputer/pinokio/releases/download/${version}/Pinokio-${version}.dmg";
      hash = "sha256-Il5zaVWu4icSsKmMjU9u1/Mih34fd+xNpF1nkFAFFGo=";
    };
    x86_64-linux = {
      url = "https://github.com/pinokiocomputer/pinokio/releases/download/${version}/Pinokio-${version}.AppImage";
```

File: pkgs/by-name/pi/pinokio/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  }
+else
+  appimageTools.wrapType2 {
+    inherit pname version src meta;
+
+    extraInstallCommands = ''
+      mkdir -p $out/share/${pname}
+      cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
+      cp -a ${appimageContents}/usr/share/icons $out/share/
+      install -Dm 444 ${appimageContents}/${pname}.desktop -t $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/by-name/pi/pinokio/package.nix:58]:
```suggestion
      mkdir -p $out/share/pinokio
      cp -a ${appimageContents}/{locales,resources} $out/share/pinokio
      cp -a ${appimageContents}/usr/share/icons $out/share/
      install -Dm 444 ${appimageContents}/pinokio.desktop -t $out/share/applications
```

**Review by SuperSandro2000 — COMMENTED** [review]
We don't want to replace any string that happily matches pname

**Pull request merged**